static.emf/public/emf_tool.json

```diff
@@ -1,111 +1,113 @@
-        "id": "emf",
-
-        "name": "Emf",
-
-        "version": "0.0.1",
-        
-        "author": "MDENet",
-        
-        "homepage": "https://github.com/mdenet/platformtools",
-
-        "functions": [
-
-            {
-
-                "id": "function-ecoretodiagram",
-
-                "name": "Conversion - Ecore To Diagram",
-
-                "parameters": [ {"name":"input", "type":"ecore"} ],
-
-                "returnType": "diagram",
-
-                "path": "http://"
-
-            },
-
-            {
-
-                "id": "function-xmitodiagram",
-
-                "name": "Conversion - Xmi To Diagram",
-
-                "parameters": [ {"name":"input", "type":"xmi"} ],
-
-                "returnType": "diagram",
-
-                "path": "http://"
-
-            }
-        ],
-
-
-
-        "panelDefs": [
-
-            {
-
-                "id": "ecore",
-
-                "name": "Ecore",
-
-                "panelclass": "ProgramPanel",
-
-                "icon": "ecore",
-
-                "language": "ecore",
-
-                "buttons" : [ 
-                      { 
-                        "id": "refresh-button", 
-                        "icon": "refresh",
-                        "renderfunction": "function-ecoretodiagram",
-                        "hint": "Render the metamodel class diagram" 
-                      }, 
-                      { 
-                        "id": "help-button", 
-                        "icon": "info",
-                        "url": "https://www.eclipse.org/modeling/emf/",
-                        "hint": "EMF Homepage" 
-                      } 
-                ]
-
-            },
-
-            {
-
-                "id": "xmi",
-
-                "name": "XMI",
-
-                "panelclass": "ProgramPanel",
-
-                "icon": "xmi",
-
-                "language": "xmi",
-
-                "buttons" : [ 
-                    { 
-                      "id": "refresh-button", 
-                      "icon": "refresh",
-                      "renderfunction": "function-flexmitograph",
-                      "hint": "Render the model object diagram" 
-                    }, 
-                    { 
-                      "id": "help-button", 
-                      "icon": "info",
-                      "url": "https://www.omg.org/spec/XMI/",
-                      "hint": "OMG - XML Metadata Interchange" 
-                    }          
-                ]
-
-            }
-
-        ]
+    "tool": 
+        {
+            "id": "emf",
+    
+            "name": "Emf",
+    
+            "version": "0.0.1",
+            
+            "author": "MDENet",
+            
+            "homepage": "https://github.com/mdenet/platformtools",
+    
+            "functions": [
+    
+                {
+    
+                    "id": "function-ecoretodiagram",
+    
+                    "name": "Conversion - Ecore To Diagram",
+    
+                    "parameters": [ 
```

Comment:
This changes the function from a conversion function to an action function. Conversion functions must have an "input" parameter and an optional "metamodel" parameter, I can't seem to find this documented in the wiki anywhere. 

Though we probably want to introduce this change to support the current way we are handling diagrams via action functions e.g. emfatic2plantuml. I think originally the intention was to implement output type conversion for action functions  conversion functions similar to that done on input parameters.

Reply:
Point taken. Output conversion functions would be nice indeed, but given that we haven't got them yet, this seems like the pragmatic solution for the time being...